### PR TITLE
Multitarget tests

### PR DESCRIPTION
**Pull request by timbussmann**

### CODE REVIEW

File: src/PerformanceTests/Tests/Categories/Base.cs [hunk 1]
```diff
@@ -10,13 +10,22 @@ namespace Categories
...
-        static readonly bool InvokeEnabled = bool.Parse(ConfigurationManager.AppSettings["InvokeEnabled"]);
-        static readonly TimeSpan MaxDuration = TimeSpan.Parse(ConfigurationManager.AppSettings["MaxDuration"]);
+        static readonly bool InvokeEnabled;
+        static readonly TimeSpan MaxDuration;
+
+        static Base()
+        {
+            // we need to get the correct config file, as .net core will try to load nunits test harness configuration file
+            // see https://github.com/dotnet/corefx/issues/22101
+            var assemblyLocation = Assembly.GetExecutingAssembly().Location;
```
Comment (danielmarbach) [inline src/PerformanceTests/Tests/Categories/Base.cs]: I was wondering if it wouldn't be simpler to replace the config file with a code first approach. @ramonsmits how often are these values changed?

File: src/PerformanceTests/Tests/Categories/Base.cs [hunk 2]
```diff
@@ -92,23 +101,27 @@ static void LaunchAndWait(Permutation permutation, TestDescriptor testDescriptor
...
+                // ReSharper disable once RedundantAssignment
+                var processIdArgument = string.Empty;
+#if NET452
+                var processId = VisualStudioDebugHelper.DebugAttacher.GetCurrentVisualStudioProcessId();
+                processIdArgument = processId >= 0 ? $" --processId={processId}" : string.Empty;
+#endif
                 pi = new ProcessStartInfo(testDescriptor.ProjectAssemblyPath, permutationArgs + sessionIdArgument + processIdArgument)
                 {
-                    UseShellExecute = false,
+                    UseShellExecute = true,
```
Comment (timbussmann) [inline src/PerformanceTests/Tests/Categories/Base.cs]: I played around with those values again, I noticed the following:
* it doesn't seem to make a difference when running on the full framework.
* it doesn't seem to make a difference when running on .net core on a linux box
* when running the .net core tests via R#, it doesn't show the endpoint console when this is set to false.

based on that, I'd propose to stick with `true` for now unless there is some other problem caused by this?

File: src/PerformanceTests/Utils/Permutations/PermutationParser.cs [hunk 3]
```diff
@@ -10,10 +10,10 @@ public static class PermutationParser
...
         public static string ToArgs(Permutation instance)
         {
-            return "--tests:" + string.Join(Separator, instance.Tests) + 
-                   " --category:" + instance.Category +
-                   " --description:\"" + instance.Description + "\"" +
-                   " --variables:" + ToString(instance);
+            return $"--tests:\"{string.Join(Separator, instance.Tests)}\"" +
+                   $" --category:\"{instance.Category}\"" +
+                   $" --description:\"{instance.Description}\"" +
+                   $" --variables:\"{ToString(instance)}\"";
```
Comment (timbussmann) [inline src/PerformanceTests/Utils/Permutations/PermutationParser.cs]: linux can't deal with `;` separated values unless they are wrapped by "

File: src/PerformanceTests/Tests/Categories/Base.cs (hunk 1 above)
Comment (ramonsmits) [reply src/PerformanceTests/Tests/Categories/Base.cs]: I'm fine with doing a code only approach for most values, especially these two. They could just be static readonly.

Comment (ramonsmits) [reply src/PerformanceTests/Tests/Categories/Base.cs]: Well, `InvokeEnabled` was used by the buildserver. Just make it a code var. Build would need to be addressed anyway as it is currently broken.

**Comment by timbussmann** [comment]
not sure removing the config manager code makes this much easier since it's working now. Happy to remove it though ;)

File: src/PerformanceTests/Tests/Categories/Base.cs [hunk 4]
```diff
@@ -92,23 +101,27 @@ static void LaunchAndWait(Permutation permutation, TestDescriptor testDescriptor
             var permutationArgs = PermutationParser.ToArgs(permutation);
             var sessionIdArgument = $" --sessionId={SessionId}";
 
+            //var useShellExecute = Environment.OSVersion.Platform != PlatformID.Win32NT;
```
Comment (danielmarbach) [inline src/PerformanceTests/Tests/Categories/Base.cs]: Should this be 🔥 ?

File: src/PerformanceTests/Tests/FodyWeavers.xml [hunk 5]
```diff
@@ -1,4 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Weavers>
-  <Caseless />
```
Comment (danielmarbach) [inline src/PerformanceTests/Tests/FodyWeavers.xml]: Why was this used and how do we know we can remove this now?

**Comment by danielmarbach** [comment]
Made two comments but they can be addressed also after the PR is merged

File: src/PerformanceTests/Tests/Categories/Base.cs (hunk 4 above)
Comment (timbussmann) [reply src/PerformanceTests/Tests/Categories/Base.cs]: yes, will do

File: src/PerformanceTests/Tests/FodyWeavers.xml (hunk 5 above)
Comment (timbussmann) [reply src/PerformanceTests/Tests/FodyWeavers.xml]: I haven't seen any explicit need for this. I checked the code for string comparisons using `==` and I found one potentially "vulnerable" case which I addressed in https://github.com/Particular/EndToEnd/pull/308

**Pull request merged**